realm/realm-library/src/objectServer/java/io/realm/SyncCredentials.java

```diff
@@ -110,6 +112,30 @@ public static SyncCredentials jwt(String jwtToken) {
         return new SyncCredentials(jwtToken, IdentityProvider.JWT, null);
     }
 
+    /**
+     * Creates credentials anonymously.
+     *
+     * @return a set of credentials that can be used to log into the Object Server using
+     * {@link SyncUser#loginAsync(SyncCredentials, String, SyncUser.Callback)}.
+     */
+    public static SyncCredentials anonymous() {
+        return new SyncCredentials("", IdentityProvider.ANONYMOUS, null);
+    }
+    /**
+     * Creates credentials using a nickname.
+     *
+     * @param nickname that identifies a user
+     * @return a set of credentials that can be used to log into the Object Server using
+     * {@link SyncUser#loginAsync(SyncCredentials, String, SyncUser.Callback)}.
+     * @throws IllegalArgumentException if the nickname is either {@code null} or empty.
+     */
+    public static SyncCredentials nickname(String nickname, boolean isAdmin) {
```

Comment:
It seems insane that it is possible to create an admin user this way? Anyone can become admin this way?

Reply:
Maybe make it very clear that his provider should be disabled for any production app?

Reply:
I had the same thought, but this provider should only be activated in dev/demo environment, not in prod. will add a note